From 086eb4573792125869749a89ef7929ceb5bca2d7 Mon Sep 17 00:00:00 2001
From: RenJian <jian.ren@ck-telecom.com>
Date: Wed, 15 Apr 2015 11:32:15 +0800
Subject: [PATCH 86/95] Ensure there is no two same storages showing on the
 computer.

[Preconditions]
1. Insert a SIM card into phone and set the SIM lock as "on"
2. Select MTP mode
3. Power off the mobile

[Procedures]
1.Connected the phone and PC with usb cable
2.Power on the phone->Input PIN code of SIM lock to enter the IDLE view
3.Check the storage list on PC

[Reproduce]
Rarely

Change-Id: I8efc3f812b669f2d4e2c3be89e3f97b5cc895628
Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 media/mtp/MtpServer.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media/mtp/MtpServer.cpp b/media/mtp/MtpServer.cpp
index 6a334e6..f9bb750 100644
--- a/media/mtp/MtpServer.cpp
+++ b/media/mtp/MtpServer.cpp
@@ -120,6 +120,8 @@ MtpServer::~MtpServer() {
 void MtpServer::addStorage(MtpStorage* storage) {
     Mutex::Autolock autoLock(mMutex);
 
+    if(hasStorage(storage->getStorageID()))
+        return;
     mStorages.push(storage);
     sendStoreAdded(storage->getStorageID());
 }
-- 
2.5.0

